metadata/metadata.go

```diff
@@ -2149,12 +2149,14 @@ type SourceVariantResource struct {
 	Inputs         []NameVariant                           `json:"inputs"`
 }
 
+// Why do we even use this?
```

Comment:
@simba-git, this is used in a shallow map function (see `SourceShallowMap` in `metadata.go`), which, oddly, appears to have an duplicate implemented in `client.go`: see [here](https://github.com/featureform/featureform/blob/main/metadata/client.go#L2180)